sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1048,6 +1067,11 @@ _pi_queue::_pi_queue(std::vector<ze_command_queue_handle_t> &ComputeQueues,
     ComputeQueueGroup.LowerIndex = FilterLowerIndex;
     ComputeQueueGroup.UpperIndex = FilterUpperIndex;
     ComputeQueueGroup.NextIndex = ComputeQueueGroup.LowerIndex;
+    // Create space to hold immediate commandlists corresponding to the ZeQueues
+    if (UseImmediateCommandLists) {
+      ComputeQueueGroup.ImmCmdLists = std::vector<pi_command_list_ptr_t>(
+          ComputeQueueGroup.ZeQueues.size(), CommandListMap.end());
+    }
```

Comment:
Force "0" index only for immediate-command lists.

Reply:
If it is forced to 0 there will need to be special provision to make it non-zero so the L0 team can investigate why non-zero indices fail. What is the issue with saying that currently because non-zero indices expose a problem in L0, SYCL_PI_LEVEL_ZERO_USE_COMPUTE_ENGINE=0 is needed?

Reply:
It is already known why it fails (and they can use env var to verify it's fixed). We need stable behavior by default. And, again, for immediate command-lists (unlike regular) it yet needs to be proven that submitting to multiple engines is preferable, so let's have a separate tuning for them (both "0" for now).

Reply:
Copy queues also have problems. Shall we set both compute and copy to 0 by default and let the env var be used to select otherwise?

Reply:
OK, set both to "0" by default for immediate submissions. But don't create new env vars, existing ones, if specified, would override the default

Reply:
OK